src/map/unit.cpp

```diff
@@ -1938,8 +1942,17 @@ void unit_set_attackdelay(block_list& bl, t_tick tick, e_delay_event event)
 	// When setting delays, we need to make sure not to decrease them in case they've been set by another source already
 	if (attack_delay > 0)
 		ud->attackabletime = i64max(tick + attack_delay, ud->attackabletime);
-	if (act_delay > 0)
-		ud->canact_tick = i64max(tick + act_delay, ud->canact_tick);
+	if (act_delay > 0) {
+		if (bl.type == BL_MOB && !(battle_config.mob_ai&0x2000)) {
+			// Monsters don't process their AI for this duration instead of having a canact_tick
+			// This is also the reason why they don't move during this time
+			mob_data& md = reinterpret_cast<mob_data&>(bl);
+			md.next_thinktime = i64max(tick + act_delay, md.next_thinktime);
+		}
+		else if(bl.type != BL_MOB) {
+			ud->canact_tick = i64max(tick + act_delay, ud->canact_tick);
+		}
```

Comment:
Please move this into the switch above and set act_delay to 0 if needed.

Reply:
Why? Here, it's the perfect place so we have the logic (and the check for the config) all in one place. Otherwise we run the risk that a developer will set canact_tick for monsters.

I also asked you if you're okay with this change before I started working on it.

Reply:
Reworked it completely not to have a clean split between attack/act delays and monster-specific delays.